src/passes/function_passes/UnoptimizedCopyPass.cpp

```diff
@@ -43,27 +44,33 @@ struct UnoptimizedCopyPass: public FunctionPass {
    * Replace calls in __unopt_ functions with calls to other __unopt_ functions.
    */
   virtual bool runOnFunction(Function &fun) {
-    if (!fun.getName().startswith(UNOPT_PREFIX)) {
-      return false;
-    }
     outs() << "Running UnoptCopyPass on function: " << fun.getName() << '\n';
-    Module *mod = fun.getParent();
-    for (auto &bb : fun) {
-      for (auto &inst : bb) {
-        if (isa<CallInst>(inst)) {
-          // replace all calls inside this unoptimized function with calls to
-          // the unoptimized versions of the called functions
-          CallInst &call = cast<CallInst>(inst);
-          Function *called_fun = call.getCalledFunction();
-          if (called_fun) {
-            StringRef called_fun_name = called_fun->getName();
-            if (!called_fun_name.startswith(UNOPT_PREFIX)) {
-              // not an unoptimized function -> must call the unoptimized
-              // version of the function instead
-              Function *new_fun =
-                mod->getFunction(UNOPT_PREFIX + called_fun_name.str());
-              if (new_fun) {
-                call.setCalledFunction(new_fun);
+    auto funName = fun.getName();
+    if (funName == TRACE_FUN_NAME || funName == "more_indirection"
```

Comment:
Similarly here, add an `XXX`?

Reply:
Can do, but it is probably best to remove everything after `||`. The only function we really **don't** want inlined is `trace`. I disabled all inlining to test my code, that's why there are some hardcoded values here and there.